### PR TITLE
Remove SerializationFactory parameter from the places where it is not needed

### PR DESCRIPTION
**Pull request by vkolomeyko**
Also minor improvements to the looping/if-branching code and documentation changes

Functionally, this change is a NOP

### CODE REVIEW

File: client/rpc/src/main/kotlin/net/corda/client/rpc/serialization/SerializationScheme.kt [hunk 1]
```diff
@@ -3,21 +3,20 @@ package net.corda.client.rpc.serialization
...
 import net.corda.core.serialization.SerializationContext
-import net.corda.core.serialization.SerializationFactory
 import net.corda.core.utilities.ByteSequence
 import net.corda.nodeapi.RPCKryo
 import net.corda.nodeapi.internal.serialization.AbstractKryoSerializationScheme
 import net.corda.nodeapi.internal.serialization.DefaultKryoCustomizer
 import net.corda.nodeapi.internal.serialization.KryoHeaderV0_1
 
-class KryoClientSerializationScheme(serializationFactory: SerializationFactory) : AbstractKryoSerializationScheme(serializationFactory) {
+class KryoClientSerializationScheme : AbstractKryoSerializationScheme() {
```
Comment (fenryka) [inline client/rpc/src/main/kotlin/net/corda/client/rpc/serialization/SerializationScheme.kt]: I know it's super fussy but Kotlin / R3 style is no space before the :

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt [hunk 2]
```diff
@@ -169,7 +153,7 @@ class GlobalTransientClassWhiteList(val delegate: ClassWhitelist) : MutableClass
 }
 
 /**
- * A whitelist that can be customised via the [CordaPluginRegistry], since implements [MutableClassWhitelist].
+ * A whitelist that can be customised via the [net.corda.core.node.CordaPluginRegistry], since implements [MutableClassWhitelist].
```
Comment (fenryka) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt]: don't think we need to fully qualify the javadoc

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt [hunk 3]
```diff
@@ -21,11 +20,7 @@ fun Kryo.addToWhitelist(type: Class<*>) {
     ((classResolver as? CordaClassResolver)?.whitelist as? MutableClassWhitelist)?.add(type)
 }
```
Comment (fenryka) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt]: javadoc still needed for class

File: client/rpc/src/main/kotlin/net/corda/client/rpc/serialization/SerializationScheme.kt (hunk 1 above)
Comment (vkolomeyko) [inline client/rpc/src/main/kotlin/net/corda/client/rpc/serialization/SerializationScheme.kt]: Hm... checked few classes like `CordaFuture` and `CordaFutureImpl` and the all have spaces before `:`

Would you be able to give an example of an extending class where `:` is used without a space?

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt (hunk 2 above)
Comment (vkolomeyko) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt]: `CordaPluginRegistry` is no longer in the imports of this class and now lives in the different sub-project so IntelliJ was giving a soft warning when package has not been fully specified for a class.

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt (hunk 3 above)
Comment (vkolomeyko) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt]: Sure - will add

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt (hunk 2 above)
Comment (fenryka) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt]: well that sucks :(

File: client/rpc/src/main/kotlin/net/corda/client/rpc/serialization/SerializationScheme.kt (hunk 1 above)
Comment (fenryka) [inline client/rpc/src/main/kotlin/net/corda/client/rpc/serialization/SerializationScheme.kt]: As with everything we're consistently inconsistent, I find best practice is to try to remember the intellij reformat code periodically :)

Comment (rick-r3) [inline client/rpc/src/main/kotlin/net/corda/client/rpc/serialization/SerializationScheme.kt]: I believe the style is: no space after variable declarations, but space before superclass/interface.

`val foo: Bar`
`class Foo : Bar`

**Comment by rick-r3** [comment]
This is no longer needed due to the removal of `KryoAMQPSerializer` I presume?

**Pull request merged**